templates/web/zurich/admin/report_edit.html

```diff
@@ -120,106 +105,131 @@
 $(function(){
-    // Show or hide the automatic reply field
+
+    $('.assignation select').change(function(){
+        if (this.value == "") {
+            $('.assignation').css('color', '#000');
+        } else {
+            var a = $(this).closest('li').css('color', '#000');
+            $('.assignation select').not(this).val("");
+            $('.assignation').not(a).css('color', '#999');
+        }
+    });
+
     $('#state').change(function(){
-        if ($(this).val() === 'hidden') {
+        // Show or hide the automatic reply field
+        var state = $(this).val();
+        if (state === 'hidden') {
             $('#automatic-reply').show();
         } else {
             $('#automatic-reply').hide();
         }
+
+        // Show or hide the categories
+        if (state === 'confirmed') {
+            $('#assignation__category').show();
+            $('#assignation__subdivision').show();
+        } else {
+            $('#assignation__category').hide();
+            $('#assignation__subdivision').hide();
+            $('#assignation__category select').val('');
+            $('#assignation__subdivision select').val('');
+        }
+
+        if ((state === 'closed') || (state === 'investigating')) {
+            $('#assignation__external').show();
+        } else {
+            $('#assignation__external select').val('');
+            $('#assignation__external').hide();
+        }
+
     }).change();
 });
 </script>
 
-[% IF problem.state == 'unconfirmed' OR problem.state == 'confirmed' %]
-
 <ul class="no-bullets">
-[% list = FOR body IN bodies %]
-    [%- NEXT UNLESS body.parent.id == c.user.from_body.id %]
-    <option value="[% body.id %]"[% IF body.id == problem.bodies_str %] selected[% END %]>[% body.name %]</option>
-[% END %]
-  [% IF admin_type != 'super' AND list %]
-    <li class="assignation">
-        <label for="body_subdivision">[% loc('Assign to subdivision:') %]</label>
-        <select name="body_subdivision" id="body_subdivision">
-            <option value="">--</option>
-            [% list %]
-        </select>
-    </li>
-  [% END %]
+[% IF problem.state == 'confirmed' %]
 
 <li class="assignation">
 <label for="category">
 [% loc('Category:') %] [% problem.category %]<br>
-[% loc('Assign to different category:') %]</label>
- <select name="category" id="category">
-        <option value="">--</option>
-      [% FOREACH cat IN categories %]
-        <option value="[% cat %]">[% cat %]</option>
-      [% END %]
-    </select></li>
 
-<li class="assignation">
-<label for="body_external">[% loc('Assign to external body:') %]</label>
-    <select name="body_external" id="body_external">
-        <option value="">--</option>
-    [% FOR body IN bodies %]
-        [% NEXT IF body.parent OR body.bodies OR body.deleted %]
-        <option value="[% body.id %]"[% IF body.id == problem.bodies_str %] selected[% END %]>[% body.name %]</option>
-    [% END %]
+[% IF problem.state == 'confirmed' %]
```

Comment:
This `IF` condition is a duplicate of the containing `IF`, so these `IF`/`END` tags can be removed.